sprocs/check_join_group.sql

```diff
@@ -0,0 +1,70 @@
+CREATE OR REPLACE FUNCTION
+    check_join_group (
+        arg_assessment_id bigint,
+        arg_user_id bigint,
+        arg_authn_user bigint,
+        arg_group_name text,
+        arg_join_code text
+    ) RETURNS void
+AS $$
+DECLARE
+    arg_group_id bigint;
+    arg_cur_size bigint;
+    arg_max_size bigint;
+BEGIN
+    -- find group id
+    SELECT 
+        g.id
+    INTO 
+        arg_group_id
+    FROM 
+        groups AS g
+        JOIN group_configs AS gc ON g.group_config_id = gc.id
+    WHERE 
+        g.name = check_join_group.arg_group_name 
+        AND g.join_code = check_join_group.arg_join_code
+        AND gc.assessment_id = check_join_group.arg_assessment_id
+        AND g.deleted_at IS NULL
+        AND gc.deleted_at IS NULL;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'Cannot find the group by provided join code';
+    END IF;
+
+    -- lock the group_users
+    PERFORM gu.group_id
+    FROM group_users AS gu
+    WHERE gu.group_id = arg_group_id
+    FOR UPDATE OF gu;
+
+    -- count the group size and compare with the max size
+    SELECT
+        COUNT(gu) AS cur_size, AVG(gc.maximum) AS maximum
```

Comment:
No need for `AS` names here.
```suggestion
        COUNT(gu), AVG(gc.maximum)
```